source/particles/particle_handler.cc

```diff
@@ -1328,15 +1308,19 @@ namespace Particles
 #ifndef DEAL_II_WITH_MPI
     (void)enable_cache;
 #else
-    // First clear the current ghost_particle information
-    for (auto &particles_in_cell : ghost_particles)
-      for (auto &particle : particles_in_cell)
-        if (particle != PropertyPool<dim, spacedim>::invalid_handle)
-          property_pool->deregister_particle(particle);
-    ghost_particles.clear();
-    ghost_particles.resize(triangulation->n_active_cells());
+    // Clear ghost particles and their properties
+    for (const auto &cell : triangulation->active_cell_iterators())
+      if (cell->is_locally_owned() == false)
+        {
+          // Clear particle properties
+          for (auto &ghost_particle : particles[cell->active_cell_index()])
+            property_pool->deregister_particle(ghost_particle);
```

Comment:
I guess this works for artificial cells as well because they never carry any particles by construction? As opposed to the other function, the negation (`!is_locally_owned()`) is not equivalent to what we have them on ghost cells.

Reply:
Very good remark. I had forgotten about this distinction. What do you think would be the best approach here? cell->is_ghost() or the current negation?
I don't think we want to keep particles in the artificial cells, it's just that they can be there initially for insertion purposes before being sorted.

Reply:
Let's wait for @gassmoeller - I would be fine merging this code as we can initially collect data, but I might miss some parts.

Reply:
Good point Martin. I would prefer `cell->is_ghost()` here. Since we agreed to not allow particle insertion into non locally owned cells there should not be any particles in artificial cells.

Reply:
Then cell->is_ghost() it is my friends!